### DISCUSSION TITLE
如何修改图片上传大小限制？

### DISCUSSION THREAD

**Discussion by musnows** [walinejs/waline#1727] [Q&A] [answered]
我使用的是hexo主题keep，今天刚刚从valine切换到了waline，使用vercel.com和leancloud部署好了之后，发现上传图片被限制到了128kb（这个大小基本没啥能传的了）我想知道怎么修改图片大小限制呢？

**Comment by lizheming** [comment, marked as answer]
https://waline.js.org/reference/client/props.html#imageuploader
https://github.com/walinejs/waline/blob/main/packages/client/src/config/default.ts#L20-L30

**Reply by musnows** [reply]
> https://waline.js.org/reference/client/props.html#imageuploader https://github.com/walinejs/waline/blob/main/packages/client/src/config/default.ts#L20-L30

我找到了[keep主题内部有关waline的部分](https://github.com/XPoet/hexo-theme-keep/blob/a8b38688e4538e11e360ed6ff4f28ac0ab5b462a/layout/_partial/comment/waline.ejs)，但是这个似乎是用cdn导入的，我没有学过这个语言，自己捣鼓了一下文档，也没找到应该如何修改上传文件大小限制😥

**Reply by Mister-Hope** [reply]
你可以向主题作者提issue要求添加主题对waline的配置功能。主题自行调用但不暴露配置项不是我们的问题。

**Reply by musnows** [reply]
> 你可以向主题作者提issue要求添加主题对waline的配置功能。主题自行调用但不暴露配置项不是我们的问题。

明白了，谢谢！

**Reply by tovarsh** [reply]
去除主题的waline cdn, 参考这篇文章自定义引入 [waline与兰空图床](https://blog.thatcoder.cn/Waline%E4%B8%8ELsky%E5%85%B0%E7%A9%BA%E5%9B%BE%E5%BA%8A/)

**Comment by musnows** [comment]
看到有其他人提这个问题，在这里提供一个解决方案：修改hexo主题导入的`waline.js`文件，在里面全局搜索128，把窗口提示符里面的`128kb`和限制里面的`128e3`中的128都改掉，就能取消128kb的上传限制（只有两处）

经过我的测试，建议设置成3mb以下，避免图片base64太大而导致页面评论加载缓慢

https://github.com/musnows/musnows.github.io/blob/master/css/my/waline.js

>个人认为128kb太小， 至少要设置成默认的512kb，才能让用户上传一部分截图；这样就不需要引入外部图床了

**Reply by Mister-Hope** [reply]
3mb base64后是4mb，假设首屏评论10条含三张的话，数据也有12mb。

本身还是鼓励用图床，而不是扔到评论里。

**Reply by Mister-Hope** [reply]
所以基于评论加载速度优先的考虑，默认就是128kb，尽可能将评论初始数据的大小控制在1m以内。
